packages/blockchain/src/consensus/clique.ts

```diff
@@ -31,7 +32,13 @@ export const CLIQUE_NONCE_DROP = Buffer.alloc(8)
 /**
  * This class encapsulates Clique-related consensus functionality when used with the Blockchain class.
  */
-export class CliqueConsensus extends ConsensusBase {
+export class CliqueConsensus implements Consensus {
+  blockchain: Blockchain
+  db: LevelUp
+  dbManager?: DBManager
+  _common?: Common
+  _validateConsensus?: boolean
```

Comment:
This `_validateConsensus` should not be passed over here but just "count" for the outer `Blockchain` class, so this guard should only take place from `Blockchain`.

Reply:
We should generally see that we also align the constructors of the different consensus classes as much as we can, optimally completely. `blockchain`, `db` and `common` can be passed in on all I guess (even if not used on some). Maybe rather also transfer to taking in an options dict as we do on other libraries (together with a respective `ConsensusOptions` interface), then we can easier expand here later on.

The `dbManager` shouldn't passed in here at all but the functionality should just be extracted (to some minor extend: replicated, but in a simpler form), see this passage from the task description:

Special note on the Clique DB functionality

> There is some clique specific DB functionality in db/manager.ts. This let me scratch my head a bit at first but on a second look: this can just be taken out from there and included into the new consensus module. The DB manager is generally a bit over-engineered and this is not much functionality related specifically to Clique (side note: and this is also not relevant regarding the DB cache, not much data).

Let me know if you need some help here, you can generally just simplify a lot here (on this complicated key <-> constant mapping with things like `CLIQUE_SIGNER_STATES_KEY` or the like, no need to keep) and then move the few remaining lines to an internal function or something.

Reply:
I switched the various consensus classes to take in just the `blockchain` instance in an options dict. That way, each consensus class has full access to the `blockchain` instance and all its various components (`db`, `common`, etc.). Seems pretty clean.

I had removed the clique-related code from `db/manager.ts`, but didn't realize there was clique-related stuff in `operations` and `constants`. I removed that in the new PR and simplified the db approach in `consensus/clique`, so it's not using `dbManager` anymore.